synapse/api/auth.py

```diff
@@ -775,15 +775,24 @@ def check_in_room_or_world_readable(self, room_id, user_id):
             )
 
     @defer.inlineCallbacks
-    def check_auth_blocking(self):
+    def check_auth_blocking(self, user_id=None):
         """Checks if the user should be rejected for some external reason,
         such as monthly active user limiting or global disable flag
+
+        Args:
+            user_id(str): If present, checks for presence against existing MAU cohort
```

Comment:
I think it'd be clearer to say something like:

```
user_id(str|None): The user ID of the requester, or None if registering.
```

So that the docstring doesn't require readers to understand what the MAU cohort is to know roughly whats going on